pkg/apis/application/v1alpha1/types.go

```diff
@@ -1484,7 +1484,13 @@ func (r ResourceResults) Find(group string, kind string, namespace string, name
 func (r ResourceResults) PruningRequired() (num int) {
 	for _, res := range r {
 		if res.Status == synccommon.ResultCodePruneSkipped {
-			num++
+			// the same "PruneSkipped" status is used for both the case that a resource required pruning
+			// and was not pruned because the sync operation had pruning disabled, and the case that a
+			// resource was skipped from pruning due to its `Prune=false` annotation, so would never need pruning
+			// so should not count towards the total number of resources that require pruning
+			if res.Message != "ignored (no prune)" {
+				num++
+			}
```

Comment:
It seems better to have a separate status. Relying on the message is unreliable.

Reply:
As `"ignored (no prune)"` is a [hard-coded string in gitops-engine](https://github.com/argoproj/gitops-engine/blob/88c35a9acf6991acc67d6da5111de8c628b4a0be/pkg/sync/sync_context.go#L1024
), it's not likely to change?

Also, it's not clear how we could change the [`ResourceResult`](https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/types.go#L1436-L1459) object to handle this specific case, because anything else would be a breaking change for the CLI.

As we would need to introduce a new [`synccommon.ResultCode`](https://github.com/argoproj/gitops-engine/blob/master/pkg/sync/common/types.go#L93-L100), and there is not a lot of room for a new one that is specific to "Prune Disabled".

---

The current behavior is a longstanding bug (https://github.com/argoproj/argo-cd/issues/8134) which hurts people's ability to automate ArgoCD with the CLI (it messes with the exit code), so I think this fix is worth it.